lib/booking_bug/connection.rb

```diff
@@ -0,0 +1,62 @@
+class BookingBug
+  class InvalidResponse < StandardError; end
+
+  class Connection
+    def login
+      parse(
+        conn.post do |req|
+          req.url '/api/v1/login'
+          req.headers['App-Id'] = app_id
+          req.headers['App-Key'] = api_key
+          req.body = credentials
+        end
+      )
+    end
+
+    def page(url, auth_token)
+      parse(
+        conn.get do |req|
+          req.url url
+          req.params['per_page'] = 100
+          req.headers['App-Id'] = app_id
+          req.headers['App-Key'] = api_key
+          req.headers['Auth-Token'] = auth_token
+        end
+      )
+    end
+
+    private
+
+    def parse(response)
```

Comment:
There is a Faraday middleware that will automatically parse JSON response types.